### PR TITLE
JEP-210 integration testing

### PR DESCRIPTION
**Pull request by jglick**
For more realistic testing, and to deal with what I presume would have been PCT failures had anyone been looking for them…?

### CODE REVIEW

**Comment by jglick** [comment]
> an incremental version of `workflow-api` as an integration test

Yes, though that would mean that this is not independently mergeable.

File: pom.xml [hunk 1]
```diff
@@ -77,12 +77,12 @@
...
             <groupId>org.jenkins-ci.plugins.workflow</groupId>
             <artifactId>workflow-step-api</artifactId>
-            <version>2.13</version>
+            <version>2.16</version>
         </dependency>
         <dependency>
             <groupId>org.jenkins-ci.plugins.workflow</groupId>
             <artifactId>workflow-api</artifactId>
-            <version>2.30</version>
+            <version>2.31-rc852.4e8faa4117f6</version> <!-- TODO https://github.com/jenkinsci/workflow-api-plugin/pull/81 -->
```
Comment (dwnusbaum) [inline pom.xml]: Hmm, this is what I have locally, but the tests still don't pass for me (not even intermittently). Maybe a subtle difference in IO behavior on macOS, or my machine is just too slow or something.

Comment (jglick) [reply pom.xml]: I will look into it.

Comment (jglick) [reply pom.xml]: I got a flake: one failure, one pass. Possibly something not fully honoring `flush`. Investigating.

Comment (dwnusbaum) [reply pom.xml]: `TaskListenerDecoratorTest#smokes` is passing consistently for me after fa173a6.

**Comment by jglick** [comment]
BTW I have searched for other `LineTransformationOutputStream` subclasses which made the same mistake and found some.

File: pom.xml [hunk 2]
```diff
@@ -124,25 +124,25 @@
...
             <artifactId>workflow-scm-step</artifactId>
-            <version>2.4</version>
+            <version>2.7</version>
             <scope>test</scope>
         </dependency>
         <dependency>
             <groupId>org.jenkins-ci.plugins.workflow</groupId>
             <artifactId>workflow-durable-task-step</artifactId>
-            <version>2.11</version>
+            <version>2.26-rc789.fba882266ed3</version> <!-- TODO https://github.com/jenkinsci/workflow-durable-task-step-plugin/pull/85 -->
```
Comment (jglick) [inline pom.xml]: https://github.com/jenkinsci/workflow-durable-task-step-plugin/pull/85 merged but as yet unreleased.

File: pom.xml [hunk 3]
```diff
@@ -198,13 +198,19 @@
...
             <groupId>org.jenkins-ci.plugins</groupId>
             <artifactId>junit</artifactId>
-            <version>1.6</version>
+            <version>1.26.1</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.jenkins-ci.plugins</groupId>
+            <artifactId>timestamper</artifactId>
+            <version>1.8.11-rc430.c45d6cdd7237</version> <!-- TODO https://github.com/jenkinsci/timestamper-plugin/pull/25 -->
```
Comment (jglick) [inline pom.xml]: https://github.com/jenkinsci/timestamper-plugin/pull/25 as yet unmerged.

**Pull request merged**